TypeUtil.isSubTypeOf may call Type#getDimensions() on non-array Type

**Describe the bug**

The TypeUtil.isSubTypeOf may call Type#getDimensions() on a non-array type, which leads to a StringIndexOutOfBoundsException when the underlying type is a primitive double, due to how asm caches primitive Types and how the array dimension algo is implemented.

**Stacktrace**

```
Exception in thread "main" org.objectweb.asm.tree.analysis.AnalyzerException: Error at instruction 10: Index 9 out of bounds for length 9
	at org.objectweb.asm.tree.analysis.Analyzer.analyze(Analyzer.java:273)
	at me.coley.analysis.SimAnalyzer.analyze(SimAnalyzer.java:74)
	at me.coley.analysis.SimAnalyzer.analyze(SimAnalyzer.java:29)
	at org.objectweb.asm.tree.analysis.Analyzer.analyzeAndComputeMaxs(Analyzer.java:297)
	at me.x150.jobf.transformers.impl.RemoveDeadCode.transformClass(RemoveDeadCode.java:101)
	at me.x150.jobf.bootstrap.Bootstrap.run(Bootstrap.java:109)
	at me.x150.jobf.Main.main(Main.java:229)
Caused by: java.lang.StringIndexOutOfBoundsException: Index 9 out of bounds for length 9
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
	at java.base/java.lang.String.checkIndex(String.java:4575)
	at java.base/java.lang.StringLatin1.charAt(StringLatin1.java:46)
	at java.base/java.lang.String.charAt(String.java:1535)
	at org.objectweb.asm.Type.getDimensions(Type.java:671)
	at me.coley.analysis.util.TypeUtil.isSubTypeOf(TypeUtil.java:164)
	at me.coley.analysis.util.TypeUtil.isSubTypeOfOrNull(TypeUtil.java:140)
	at me.coley.analysis.SimInterpreter.naryOperation(SimInterpreter.java:934)
	at me.coley.analysis.SimInterpreter.naryOperation(SimInterpreter.java:35)
	at org.objectweb.asm.tree.analysis.Frame.executeInvokeInsn(Frame.java:684)
	at org.objectweb.asm.tree.analysis.Frame.execute(Frame.java:609)
	at me.coley.analysis.SimFrame.execute(SimFrame.java:86)
	at org.objectweb.asm.tree.analysis.Analyzer.analyze(Analyzer.java:165)
	... 6 more
```


**Code location**
TypeUtil.java, line 161 - 164.

**Suggested fix**
Don't call .getDimensions() on the element type, call it on the array itself. Then get the element type when checking the actual element type in the same if later on.

--- a/src/main/java/me/coley/analysis/util/TypeUtil.java
+++ b/src/main/java/me/coley/analysis/util/TypeUtil.java
@@ -158,12 +158,12 @@ public class TypeUtil {
 		if (bothArrays) {
 			// TODO: With usage cases of "isSubTypeOf(...)" should we just check the element types are equals?
 			//  - Or should sub-typing with array element types be used like it currently is?
-			child = child.getElementType();
-			parent = parent.getElementType();
 			// Dimensions must match, unless both are Object
 			if (child.getDimensions() != parent.getDimensions() &&
-					!(child.equals(OBJECT_TYPE) && parent.equals(OBJECT_TYPE)))
+					(!child.getElementType().equals(OBJECT_TYPE) || !parent.getElementType().equals(OBJECT_TYPE)))
 				return false;
+			child = child.getElementType();
+			parent = parent.getElementType();
 		}
 		// Null check in case
 		if (parent == null)

